src/main/java/com/aws/greengrass/logmanager/LogManagerService.java

```diff
@@ -860,8 +861,14 @@ public void updateFromTopic(Topic topic) {
                     lastModifiedTime = Coerce.toLong(topic);
                     break;
                 case PERSISTED_CURRENT_PROCESSING_FILE_HASH:
-                    //TODO: the scenario of upgrading from older version needs to be handled in next PR.
+                    // If upgrade from older version, then the fileHash does not exist but fileName exists, then
+                    // transfer from fileName to fileHash.
+                    String savedName = Coerce.toString(PERSISTED_CURRENT_PROCESSING_FILE_NAME);
                     fileHash = Coerce.toString(topic);
+                    if (Objects.isNull(fileHash) && !Objects.isNull(savedName)) {
+                        LogFile savedFile = new LogFile(Paths.get(savedName).toUri());
+                        fileHash = savedFile.hashString();
+                    }
```

Comment:
this can throw an exception. Make sure that we're handling it correctly.

Reply:
Sorry, what exception? Is that something expected?

Reply:
`LogFile` is a `File`. `File`'s constructor can throw exceptions. `Paths.get()` can also throw exceptions, such as `IllegalArgumentException` these must be handled appropriately so we don't kill log manager.